test/TopupBptTest.t.sol

```diff
@@ -24,14 +24,13 @@ contract TopupBptTest is BaseFixture {
         uint256 initialEureBal = IERC20(EURE).balanceOf(address(safe));
         uint256 initialBptBal = IERC20(BPT_STEUR_EURE).balanceOf(address(safe));
 
-        (bool canExec, bytes memory execPayload) = roboModule.checker();
-        (bytes memory dataWithoutSelector, bytes4 selector) = _extractEncodeDataWithoutSelector(execPayload);
+        (bool canExec, bytes memory execPayload) = roboModule.checkUpkeep("");
         (RoboSaverVirtualModule.PoolAction _action, uint256 _amount) =
-            abi.decode(dataWithoutSelector, (RoboSaverVirtualModule.PoolAction, uint256));
+            abi.decode(execPayload, (RoboSaverVirtualModule.PoolAction, uint256));
 
         // since initially it was minted 1000 it should be way above the buffer
         assertTrue(canExec, "CanExec: not executable");
-        assertEq(selector, ADJUST_POOL_SELECTOR, "Selector: not adjust pool (0xba2f0056)");
+        // assertEq(selector, ADJUST_POOL_SELECTOR, "Selector: not adjust pool (0xba2f0056)");
```

Comment:
idem

Reply:
how do you envision extracting the selector exactly?

Reply:
this one just below has the other assert of the action